sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -189,34 +222,84 @@ export function fromRheaMessage(msg: RheaMessage): EventDataInternal {
     }
   }
 
+  if (msg.content_type != null) {
+    data.contentType = msg.content_type;
+  }
+  if (msg.correlation_id != null) {
+    data.correlationId = msg.correlation_id;
+  }
+  if (msg.message_id != null) {
+    data.messageId = msg.message_id;
+  }
+
   return data;
 }
 
 /**
  * Converts an EventData object to an AMQP message.
  * @param data - The EventData object that needs to be converted to an AMQP message.
  * @param partitionKey - An optional key to determine the partition that this event should land in.
- * @hidden
+ * @internal
  */
-export function toRheaMessage(data: EventData, partitionKey?: string): RheaMessage {
-  const msg: RheaMessage = {
-    body: data.body
-  };
-  // As per the AMQP 1.0 spec If the message-annotations or delivery-annotations section is omitted,
-  // it is equivalent to a message-annotations section containing anempty map of annotations.
-  msg.message_annotations = {};
-  if (data.properties) {
-    msg.application_properties = data.properties;
-  }
-  if (isDefined(partitionKey)) {
-    msg.message_annotations[Constants.partitionKey] = partitionKey;
-    // Event Hub service cannot route messages to a specific partition based on the partition key
-    // if AMQP message header is an empty object. Hence we make sure that header is always present
-    // with atleast one property. Setting durable to true, helps us achieve that.
-    msg.durable = true;
+export function toRheaMessage(
+  data: EventData | AmqpAnnotatedMessage,
+  partitionKey?: string
+): RheaMessage {
+  let rheaMessage: RheaMessage;
+  if (isAmqpAnnotatedMessage(data)) {
+    rheaMessage = {
+      ...AmqpAnnotatedMessage.toRheaMessage(data),
+      body: defaultDataTransformer.encode(data.body, data.bodyType ?? "data")
+    };
+  } else {
+    let bodyType: BodyTypes = "data";
+    if (typeof (data as EventDataInternal).getRawAmqpMessage === "function") {
```

Comment:
So this check is basically isReceivedEventData(), more or less?

Reply:
Right now, yes, exactly.